cmd/jaeger/internal/extension/jaegerstorage/extension.go

```diff
@@ -157,43 +157,49 @@ func (s *storageExt) Start(_ context.Context, host component.Host) error {
 	}
 	for storageName, cfg := range s.config.TraceBackends {
 		s.telset.Logger.Sugar().Infof("Initializing storage '%s'", storageName)
-		var factory storage.Factory
+		var factory tracestore.Factory
 		var err error = errors.New("empty configuration")
 		switch {
 		case cfg.Memory != nil:
-			factory, err = memory.NewFactoryWithConfig(
+			memoryFactory := memory.NewFactoryWithConfig(
 				*cfg.Memory,
 				scopedMetricsFactory(storageName, "memory", "tracestore"),
 				s.telset.Logger,
-			), nil
+			)
+			factory, err = v1adapter.NewFactory(memoryFactory), nil
 		case cfg.Badger != nil:
-			factory, err = badger.NewFactoryWithConfig(
+			badgerFactory, badgerErr := badger.NewFactoryWithConfig(
 				*cfg.Badger,
 				scopedMetricsFactory(storageName, "badger", "tracestore"),
 				s.telset.Logger)
+			factory, err = v1adapter.NewFactory(badgerFactory), badgerErr
```

Comment:
suggestion: declare `var v1factory` at the top and here only do
```
v1factory, err = badger.NewFactoryWithConfig(...)
```
then after the switch check of `v1factory != nil` then 
```
factory = v1adapter.NewFactory(v1factory)
```